File: _includes/code/llms-txt/python/named_vectors.py
"""llms.txt snippet: named vectors. Section "Python / TypeScript > Named vectors"."""
import os
import time
import weaviate
from weaviate.classes.init import Auth

client = weaviate.connect_to_weaviate_cloud(
    cluster_url=os.environ["WEAVIATE_URL"],
    auth_credentials=Auth.api_key(os.environ["WEAVIATE_API_KEY"]),
)
client.collections.delete("Article__NvPy")

# START llms_named_vectors
from weaviate.classes.config import Configure, Property, DataType

client.collections.create(
    "Article__NvPy",
    vector_config=[
        Configure.Vectors.text2vec_weaviate(name="title", source_properties=["title"]),
        Configure.Vectors.text2vec_weaviate(name="body", source_properties=["body"]),
    ],
    properties=[
        Property(name="title", data_type=DataType.TEXT),
        Property(name="body", data_type=DataType.TEXT),
    ],
)
col = client.collections.use("Article__NvPy")
res = col.query.near_text("machine learning", target_vector="title", limit=3)
# END llms_named_vectors

col.data.insert({"title": "Deep learning advances", "body": "A study of neural networks."})
time.sleep(2)
res = col.query.near_text("machine learning", target_vector="title", limit=3)
assert len(res.objects) == 1
client.collections.delete("Article__NvPy")
client.close()

File: _includes/code/llms-txt/typescript/named_vectors.ts
// llms.txt snippet: named vectors. Section "Python / TypeScript > Named vectors".
import weaviate, { vectors, dataType } from 'weaviate-client';

const client = await weaviate.connectToWeaviateCloud(process.env.WEAVIATE_URL!, {
  authCredentials: new weaviate.ApiKey(process.env.WEAVIATE_API_KEY!),
});
await client.collections.delete('Article__NvTs');

// START llms_named_vectors
await client.collections.create({
  name: 'Article__NvTs',
  vectorizers: [
    vectors.text2VecWeaviate({ name: 'title', sourceProperties: ['title'] }),
    vectors.text2VecWeaviate({ name: 'body', sourceProperties: ['body'] }),
  ],
  properties: [
    { name: 'title', dataType: dataType.TEXT },
    { name: 'body', dataType: dataType.TEXT },
  ],
});
const col = client.collections.use('Article__NvTs');
const res = await col.query.nearText('machine learning', { targetVector: 'title', limit: 3 });
// END llms_named_vectors

await col.data.insert({ title: 'Deep learning advances', body: 'A study of neural networks.' });
await new Promise((r) => setTimeout(r, 2000));
const res2 = await col.query.nearText('machine learning', { targetVector: 'title', limit: 3 });
if (res2.objects.length !== 1) throw new Error('expected 1 article');
await client.collections.delete('Article__NvTs');
await client.close();
